extension/storage/filestorage/extension.go

```diff
@@ -38,8 +39,20 @@ var _ storage.Extension = (*localFileStorage)(nil)
 
 func newLocalFileStorage(logger *zap.Logger, config *Config) (component.Extension, error) {
 	info, err := os.Stat(config.Directory)
-	if (err != nil && os.IsNotExist(err)) || !info.IsDir() {
-		return nil, fmt.Errorf("directory must exist: %v", err)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return nil, fmt.Errorf("directory must exist: %v", err)
+		}
+		if fsErr, ok := err.(*fs.PathError); ok {
+			return nil, fmt.Errorf(
+				"problem accessing configured directory: %s, err: %v",
+				config.Directory, fsErr,
+			)
+		}
+
+	}
+	if !info.IsDir() {
+		return nil, fmt.Errorf("%s is not a directory: %v", config.Directory, err)
 	}
```

Comment:
May want to have this in "validate" func on the config.

Reply:
I've changed it but that required more changes in tests because now the default config might not pass on machines that don't have the default dir created (hence my changes in this PR overriding the default in some tests).

Reply:
Any chance that we revisit this?

It seems that `load-tests / loadtest (TestBallastMemory|TestLog10kDPS) (pull_request) ` failed for some reason but I wouldn't connect it to this PR.